pkg/networkutils/network.go

```diff
@@ -1121,30 +1112,30 @@ func setupENINetwork(eniIP string, eniMAC string, deviceNumber int, eniSubnetCID
 	}
 	if err := netLink.RouteDel(&defaultRoute); err != nil {
 		if !netlinkwrapper.IsNotExistsError(err) {
-			return gw, errors.Wrapf(err, "setupENINetwork: unable to delete default route %s for source IP %s", eniSubnetIPNet.String(), eniIP)
+			return errors.Wrapf(err, "setupENINetwork: unable to delete default route %s for source IP %s", eniSubnetIPNet.String(), eniIP)
 		}
 	}
-	return gw, nil
+	return nil
 }
 
 // For IPv6 strict mode, ICMPv6 packets from the gateway must lookup in the local routing table so that branch interfaces can resolve their gateway.
-func (n *linuxNetwork) UpdateIPv6GatewayRule(gw *net.IP) error {
+func (n *linuxNetwork) createIPv6GatewayRule() error {
 	gatewayRule := n.netLink.NewRule()
-	gatewayRule.Src = &net.IPNet{IP: *gw, Mask: net.CIDRMask(128, 128)}
+	gatewayRule.Src = &net.IPNet{IP: GetIPv6Gateway(), Mask: net.CIDRMask(128, 128)}
```

Comment:
I'm curious: Was this sufficient, during your testing?

If I remember my AWS-internal discussion with the VPC folks leading up to this FE80:EC2::1 launch, the idea was that the gateway would _respond_ to these additional addresses, but still use the original SLAAC address for traffic _initiated_ from the gateway (eg router advertisements).  This was an important detail, to avoid changing any user-visible behaviour.

I'm not sure what "can resolve their gateway" includes exactly.  I _think_ what you've done in this PR is break router-initiated RAs, but (continue to) allow explicit host-initiated NDs.  If so, I think one of the following is true:
1. sgpp needs ICMPv6, and this breaks it, and we need to fix/revert this line. 👎 
2. sgpp needs ICMPv6, and works with this change.  Best answer if true, but we might need to expand this rule to cover all the other variants of gateway address ($subnet::1, original SLAAC address). 👍
3. sgpp no longer needs ICMPv6 (RAs?) now that we're hardcoding fe80:ec2::1 elsewhere, and ICMPv6 to/from gateway is broken but we don't care.

If it's this third case, I think we should remove this `createIPv6GatewayRule()` function entirely.